### PR TITLE
feat: use eaglesong as new pow

### PR DESCRIPTION
**Pull request by quake**

### CODE REVIEW

File: miner/src/miner.rs [hunk 1]
```diff
@@ -74,65 +70,64 @@ impl Miner {
...
-                    .header(header)
-                    .build()
-                    .into_view();
-                let block_hash: H256 = block.hash().unpack();
-                if self.stderr_is_tty {
-                    debug!("Found! #{} {:#x}", block.number(), block_hash);
-                } else {
-                    info!("Found! #{} {:#x}", block.number(), block_hash);
-                }
+            self.notify_workers(WorkerMessage::Stop);
```
Comment (u2) [inline miner/src/miner.rs]: I think it's an optimization.

File: miner/Cargo.toml [hunk 2]
```diff
@@ -30,7 +29,7 @@ ckb-verification = { path = "../verification" }
 indicatif = "0.11"
 console = "0.7.5"
 ckb-dao = { path = "../util/dao" }
-ckb-hash = { path = "../util/hash"}
+eaglesong = { git = "https://github.com/rink1969/eaglesong", version = "0.1" }
```
Comment (yangby-cryptape) [inline miner/Cargo.toml]: I asked @rink1969 if he could publish it, and he did. So, we can use it from [crates.io](https://crates.io/crates/eaglesong).

Comment (doitian) [reply miner/Cargo.toml]: published

File: miner/src/miner.rs [hunk 3]
```diff
@@ -23,9 +19,9 @@ pub struct Miner {
     pub works: LruCache<H256, Work>,
     pub worker_controllers: Vec<WorkerController>,
     pub work_rx: Receiver<Work>,
-    pub seal_rx: Receiver<(H256, Seal)>,
+    pub nonce_rx: Receiver<(H256, u64)>,
     pub pb: ProgressBar,
-    pub seals_found: u64,
+    pub nonces_found: u64,
```
Comment (doitian) [inline miner/src/miner.rs]: Is `nonces_found` the same as blocks found? I'm not sure which we should display in the progress, blocks found or nonces tried.

Comment (quake) [reply miner/src/miner.rs]: it's same as blocks found

File: miner/src/worker/mod.rs [hunk 4]
```diff
@@ -90,11 +90,11 @@ pub fn start_worker(
                 panic!("incompatible pow engine and worker type");
             }
         }
-        WorkerConfig::CuckooSimple(config) => {
-            if let Some(cuckoo_engine) = pow.as_any().downcast_ref::<CuckooEngine>() {
+        WorkerConfig::EaglesongSimple(config) => {
+            if let Some(_eaglesong_engine) = pow.as_any().downcast_ref::<EaglesongPowEngine>() {
```
Comment (doitian) [inline miner/src/worker/mod.rs]: If we don't need to extract the inner item, just use `is_some()`.

File: shared/src/config.rs [hunk 5]
```diff
@@ -0,0 +1,52 @@
+use numext_fixed_hash::H256;
+use numext_fixed_uint::U256;
+use ckb_core::block::Block;
+use ckb_core::header::{BlockNumber, Header, RawHeader};
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct Config {
```
Comment (doitian) [inline shared/src/config.rs]: Any reason we add this new file?

Comment (quake) [reply shared/src/config.rs]: ops, it's a rebase error...

**Pull request merged**